internal/describe/application.go

```diff
@@ -0,0 +1,105 @@
+/*
+ *  *******************************************************************************
+ *  * Copyright (c) 2019 Edgeworx, Inc.
+ *  *
+ *  * This program and the accompanying materials are made available under the
+ *  * terms of the Eclipse Public License v. 2.0 which is available at
+ *  * http://www.eclipse.org/legal/epl-2.0
+ *  *
+ *  * SPDX-License-Identifier: EPL-2.0
+ *  *******************************************************************************
+ *
+ */
+
+package describe
+
+import (
+	"fmt"
+
+	"github.com/eclipse-iofog/iofogctl/internal/config"
+	"github.com/eclipse-iofog/iofogctl/pkg/iofog/client"
+	"github.com/eclipse-iofog/iofogctl/pkg/util"
+)
+
+type applicationExecutor struct {
+	namespace string
+	name      string
+	filename  string
+	flow      *client.FlowInfo
+	client    *client.Client
+	msvcs     []client.MicroserviceInfo
+	msvcPerID map[string]*client.MicroserviceInfo
+}
+
+func newApplicationExecutor(namespace, name, filename string) *applicationExecutor {
+	a := &applicationExecutor{}
+	a.namespace = namespace
+	a.name = name
+	a.filename = filename
+	return a
+}
+
+func (exe *applicationExecutor) init(controller *config.Controller) (err error) {
+	exe.client = client.New(controller.Endpoint)
+	if err = exe.client.Login(client.LoginRequest{Email: controller.IofogUser.Email, Password: controller.IofogUser.Password}); err != nil {
+		return
+	}
+	exe.flow, err = exe.client.GetFlowByName(exe.name)
+	if err != nil {
+		return
+	}
+	msvcListResponse, err := exe.client.GetMicroservicesPerFlow(exe.flow.ID)
+	if err != nil {
+		return
+	}
+	exe.msvcs = msvcListResponse.Microservices
+	exe.msvcPerID = make(map[string]*client.MicroserviceInfo)
```

Comment:
another array to map conversion you might be able to use generic function for

Reply:
Without golang generics, there is too many small variations to make a generic array to map function